src/tir/schedule/analysis.h

```diff
@@ -132,6 +132,15 @@ void CheckReductionBlock(const ScheduleState& self, const StmtSRef& block_sref,
 bool IsAffineBinding(const BlockRealize& realize, const Map<Var, Range>& loop_var_ranges,
                      arith::Analyzer* analyzer);
 
+/*!
+ * \brief Check whether a block has an affine binding using the cached flag, and throw an exception
+ * if the block does not have an affine binding.
+ * \param self The schedule state
+ * \param block The block to be checked
+ * \throw ScheduleError If the input block does not have an affine binding
+ */
+void CheckAffineBinding(const ScheduleState& self, Block block);
```

Comment:
Given the method is not consuming the block in most of the cases

```suggestion
void CheckAffineBinding(const ScheduleState& self, const Block& block);
```